include/nuttx/syslog/syslog.h

```diff
@@ -82,22 +82,41 @@
  * Public Types
  ****************************************************************************/
 
-/* This structure provides the interface to a SYSLOG device */
+/* Forward declaration */
 
-typedef CODE ssize_t (*syslog_write_t)(FAR const char *buf, size_t buflen);
-typedef CODE int (*syslog_putc_t)(int ch);
-typedef CODE int (*syslog_flush_t)(void);
+struct syslog_channel_s;
 
-struct syslog_channel_s
-{
-  /* I/O redirection methods */
+/* SYSLOG I/O redirection methods */
+
+typedef CODE ssize_t (*syslog_write_t)(FAR struct syslog_channel_s *channel,
+                                       FAR const char *buf, size_t buflen);
+typedef CODE int (*syslog_putc_t)(FAR struct syslog_channel_s *channel,
+                                  int ch);
+typedef CODE int (*syslog_flush_t)(FAR struct syslog_channel_s *channel);
+
+/* SYSLOG device operations */
 
+struct syslog_channel_ops_s
+{
   syslog_putc_t  sc_putc;   /* Normal buffered output */
   syslog_putc_t  sc_force;  /* Low-level output for interrupt handlers */
   syslog_flush_t sc_flush;  /* Flush buffered output (on crash) */
 #ifdef CONFIG_SYSLOG_WRITE
   syslog_write_t sc_write;  /* Write multiple bytes */
 #endif
+};
+
+/* This structure provides the interface to a SYSLOG channel */
+
+struct syslog_channel_s
+{
+  /* Channel operations */
+
+  FAR const struct syslog_channel_ops_s *sc_ops;
+
+  /* Channel device */
+
+  FAR void *sc_dev;
```

Comment:
remove sc_dev, let's driver embed syslog_channel_s into his own struct directly like this:
```
struct my_channel_s
{
  struct syslog_channel_s;
 /* add more sate here */
};

static int my_syslog_putc)(FAR struct syslog_channel_s *channel,  int ch)
{
  FAR struct my_channel_s *my = (FAR struct my_channel_s *)channel;

  /* access the state by my->xxx */
}
```
This is the convention used by many driver.

Reply:
@xiaoxiang781216 Please have a look whether it is OK now, so I can proceed with testing it.